### PR TITLE
Upgrade ui

### PR DESCRIPTION
**Pull request by jojohappy**
## Changes

fixes: #1042 
Upgrade the Thanos ui to the Prometheus 2.9.1, new style of ui theme would be used, although I don't like it.

## Verification

To visit the Thanos Query and Rule page.

@adrien-f Please help me to verify the `store` page, you are the original committer, thanks!

@bwplotka @povilasv Please help me to verify all of the pages, especially the `rule` page, I have not a rule component, thanks!

### CODE REVIEW

**Comment by povilasv** [comment]
@jojohappy  nice work! 

Add an entry to chanelog please :)

Do you have a docker image in dockerhub with this commit so that I can play around with it?

**Comment by bwplotka** [comment]
Nice!

**Comment by jojohappy** [comment]
@povilasv @bwplotka @adrien-f you can get the image by `jojohappy/thanos:upgrade_ui`. Thanks!

**Comment by GiedriusS** [comment]
Nice but it's kind of hard to review so many different (generated) files. Could you tell us how did you do this? Did you copy/paste over the relevant files? 😄

File: pkg/ui/static/js/graph_template.handlebar [hunk 1]
```diff
@@ -80,14 +74,14 @@
...
 
                           <button
-                              class="btn btn-default pull-left"
+                              class="btn btn-light pull-left"
                               type="button"
                               name="dec_end"
                               title="Rewind the end time.">
                             <i class="glyphicon glyphicon-backward"></i>
-                          </button>
+                          </button><!--
```
Comment (GiedriusS) [inline pkg/ui/static/js/graph_template.handlebar]: do we need this comment, for example? that's why I'm wondering 😄

File: pkg/ui/templates/graph.html [hunk 2]
```diff
@@ -15,15 +13,21 @@
 
     <script src="{{ pathPrefix }}/static/vendor/mustache/mustache.min.js?v={{ buildVersion }}"></script>
     <script src="{{ pathPrefix }}/static/vendor/js/jquery.selection.js?v={{ buildVersion }}"></script>
-    <script src="{{ pathPrefix }}/static/vendor/js/jquery.hotkeys.js?v={{ buildVersion }}"></script>
+    <!-- <script src="{{ pathPrefix }}/static/vendor/js/jquery.hotkeys.js?v={{ buildVersion }}"></script> -->
```
Comment (GiedriusS) [inline pkg/ui/templates/graph.html]: why is this commented out too, for example?

**Comment by jojohappy** [comment]
@GiedriusS Maybe we could play around this PATCH by the image `jojohappy/thanos:upgrade_ui` (: 
Because we are following the Prometheus ui, as you said, I copied and pasted the relevant files and upgraded the frontend vendors for bootstrap 4.0. Exclude the `store` page, it wrote by @adrien-f .

File: pkg/ui/static/js/graph_template.handlebar (hunk 1 above)
Comment (jojohappy) [reply pkg/ui/static/js/graph_template.handlebar]: Just following the upstream (:  I don't know why it was not deleted by Prometheus too. ):

File: .gitignore [hunk 3]
```diff
@@ -1,6 +1,7 @@
 /prometheus
 /thanos
 vendor/
+!pkg/ui/static/vendor
```
Comment (bwplotka) [inline .gitignore]: Let's change the upper vendor to `/vendor/` then instead of this ?

**Comment by bwplotka** [comment]
@povilasv @jojohappy, I built: `improbable/thanjojohappy-ui-upgrade1`

**Comment by bwplotka** [comment]
Are we sure the security patch for query history issue is there as well?

**Comment by jojohappy** [comment]
> Are we sure the security patch for query history issue is there as well?
@bwplotka I don't know what security patch you said, but I searched the list of prometheus issues, is this [issue](https://github.com/prometheus/prometheus/issues/3789) you mentioned?

**Comment by GiedriusS** [comment]
I'd love to continue with this PR since it would solve an important pain point. @jojohappy, could you please point me from which exact commit in https://github.com/prometheus/prometheus you've pulled this from?

**Comment by povilasv** [comment]
I've tested Thanos Rule and I get `template: :64: function "since" not defined` when opening `/rules` page.

**Comment by povilasv** [comment]
apart from that it looks awesome

**Comment by jojohappy** [comment]
> I'd love to continue with this PR since it would solve an important pain point. @jojohappy, could you please point me from which exact commit in https://github.com/prometheus/prometheus you've pulled this from? I'd love to double check.

Thanks @GiedriusS ! I followed the ui of Prometheus v2.9.1: [commit](https://github.com/prometheus/prometheus/commit/ad71f2785fc321092948e33706b04f3150eee44f).

**Comment by jojohappy** [comment]
@povilasv @GiedriusS I have updated the image for this PR: `jojohappy/thanos:upgrade_ui` and fixed the missing template function. PTAL, thanks!

**Comment by adrien-f** [comment]
Awesome work 🎉 !

**Comment by povilasv** [comment]
:1st_place_medal:  amazing

**Comment by bwplotka** [comment]
Thanks for massive work on this @jojohappy !! Really great help :heart:

**Comment by povilasv** [comment]
:heart: @jojohappy

**Pull request merged**